GooglePlayInstant/LoadingScreen/LoadingScreen.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.SceneManagement;
+using UnityEngine.UI;
+
+namespace GooglePlayInstant.LoadingScreen
+{
+    /// <summary>
+    /// Downloads the AssetBundle available at AssetBundleUrl and updates LoadingBar with its progress
+    /// </summary>
+    public class LoadingScreen : MonoBehaviour
+    {
+        [Tooltip("The url used to fetch the AssetBundle on Start")]
+        public string AssetBundleUrl;
+
+        [Tooltip("The LoadingBar used to indicated download and install progress")]
+        public LoadingBar LoadingBar;
+
+        public RawImage Background;
+
+        private const int MaxAttemptCount = 3;
+        private AssetBundle _bundle;
+        private int _assetBundleRetrievalAttemptCount;
+
+        private IEnumerator Start()
+        {
+            while (_bundle == null && _assetBundleRetrievalAttemptCount < MaxAttemptCount)
+            {
+                yield return GetAssetBundle(AssetBundleUrl);
+            }
+
+            if (_bundle == null)
+            {
+                Debug.LogError("AssetBundle failed to be downloaded.");
+                yield break;
```

Comment:
I wonder if we should have a button for this case? Maybe a TODO for now:

// TODO: Develop UI for when AssetBundle download fails, e.g. a button to ShowInstallPrompt or exit the game?

Reply:
This would be where we would show a download error with a retry button. A TODO is a good idea for now though.

Reply:
Added `// TODO: Develop UI for when AssetBundle download fails, e.g. a user prompt with a retry button.`